init migrationToNextra

kjhahjdhjkadsf opened the pull request:
1. Link to the GitHub location of each page
2. Support multi-lingual documentation
3. Implement an auto-translation service to reduce the manual overhead of translations
4. Introduce a search feature to the documentation
5. Update documentation to support user feedback

denverbaumgartner commented:
@kjhahjdhjkadsf thanks for submitting this RP and for your efforts so far on this project! upon initial review, here are some things that stand out: 

- [ ] **page(overview):** it looks like we have lost the ability to navigate to the pages referenced by each icon / section, along with some degraded formatting. ideally this maintains parity (in style and navigability) with the [current overview page](https://docs.rubicon.finance/docs/introduction/overview)
- [ ] **page(python sdk):** same issue as addressed above, looks like linking functionality was lost to external sites
- [ ] **feature(link to relevant GH on each page):** looks like all of these link to the correct location, it would be great if this feature was more visible to the user. a great example of this is how the Graph does this by keeping the link and GH icon at the top of the page menu on the right of every page, here is an [example](https://thegraph.com/docs/en/).
- [ ] **feature(feedback):** great that this links directly to the GH repo issues, but a bit different from the goal outlined in the proposal feedback (section 1. Code-Update(s), pt. 5) [here](https://forum.rubicon.finance/t/grant-proposal-easy-contribution-guidelines-for-rubicon-documentation/69/2?u=denver). You can see an example of this from the Ethereum Foundation at the bottom of the page [here](https://ethereum.org/en/developers/docs/).
- [ ] **style(language button):** this is perfect functionally, i think its not the most obvious and users would benefit from it being located clearly at the top, the Graph documentation provides a great example [here](https://thegraph.com/docs/en/).
- [ ] **feature(language support):** great work on this, looks like auto-translation is set up and ready to go. for global coverage, i think we should support the following languages by default: 1) english (en), 2) chinese (zh), 3) hindi (hi), 4) spanish (es), 5) french (fr), 6) arabic (ar), 7) russian (ru), 8) japanese (ja), 9) korean (ko), 10) vietnamese, 11) persian (fa), 12) indonesian (id)
    - [ ] if you could provide a brief write-up of the translation pipeline you are using along with any plans for continued integration that would be greatly appreciated
    
again, kudos and the job well done. please let me know if you have any questions or concerns with the feedback provided above.

kjhahjdhjkadsf commented:
@denverbaumgartner 

Thank you for your detailed feedback on the project! I appreciate your time and effort in reviewing the work. Please find my responses to your comments below:

1 & 2. I have fixed the navigation and linking issues on the overview and Python SDK pages. They should now maintain parity with the current overview page and provide the desired functionality.

3. I have moved the subsections on the page (aka "On This Page") to the left menu, which should improve the visibility and organization of the content even further. After making this change, the current placement of the GitHub link on each page is just perfect. 

4. While I understand the proposal feedback's intent, having GitHub Issues directly linked is more beneficial and transparent. It allows the team and the community to contribute and work on any identified issues. I hope this approach matches your expectations.

5. I haven't had the chance to work on relocating the language button yet. Please let me know if this is still a priority, and I will make the necessary changes.

6. I have added the requested language support, and the auto-translation feature is now set up for global coverage.

6.1 As for the translation pipeline, there is a folder in the check-in called "translator-copy-pages" which contains a guide and the script itself. While it could be better, it works for the current scope. We can continue to refine and integrate this pipeline as needed.

Thank you once again for your valuable feedback, and please feel free to reach out if you have any further questions or concerns. I'm looking forward to continuing our collaboration on this project!

denverbaumgartner commented:
@kjhahjdhjkadsf this is excellent! Thank so much for the speedy response and iterations based upon the feedback provided.  As you addressed above, then only thing that remains in my eyes is: 

- [ ] moving both the language toggle and the light/dark mode selector to the top menu bar. i believe these features could go in between the search bar and the github link, but am open to other suggestions. the main goal is making it clear to a user that they can toggle both the language and dark/light mode. 

after this is completed i would consider this portion of the project complete and ready for merging. @kjhahjdhjkadsf it has been a pleasure working with you so far and I look forward to our continued collaboration!

bghughes commented:
> @kjhahjdhjkadsf thanks for the quick update, this looks good to me! let me pass to @bghughes for the final review, then i would consider this ready to be merged! excellent work 🚀

Thank you and excellent work @kjhahjdhjkadsf - the only change I would like to see is adding the new deployments pages as I mentioned in discord AND fixing the merge conflict here :)

kjhahjdhjkadsf commented:
@bghughes do you have a list of pages for the merge?
